functions/local_functions_test.go

```diff
@@ -0,0 +1,233 @@
+			if result.expectedError == "" {
+				require.NoError(t, err)
+				require.NotNil(t, localVariant)
+			} else {
+				require.Error(t, err)
+				assert.ErrorContains(t, err, result.expectedError)
+			}
+		})
+	}
+}
+
+var aggregateFunctionDialectYaml = `
+name: test
+type: sql
+dependencies:
+  arithmetic:
+    https://github.com/substrait-io/substrait/blob/main/extensions/functions_arithmetic.yaml
+  arithmetic_decimal:
+    https://github.com/substrait-io/substrait/blob/main/extensions/functions_arithmetic_decimal.yaml
+  comparison:
+    https://github.com/substrait-io/substrait/blob/main/extensions/functions_comparison.yaml
+  string:
+    https://github.com/substrait-io/substrait/blob/main/extensions/functions_string.yaml
+supported_types:
+  i8:
+    sql_type_name: INTEGER
+    supported_as_column: true
+  i16:
+    sql_type_name: INTEGER
+    supported_as_column: true
+  i32:
+    sql_type_name: INTEGER
+    supported_as_column: true
+  i64:
+    sql_type_name: BIGINT
+    supported_as_column: true
+  fp64:
+    sql_type_name: DOUBLE
+    supported_as_column: true
+  decimal:
+    sql_type_name: NUMBER
+    supported_as_column: true
+aggregate_functions:
+  - name: arithmetic.sum
+    aggregate: true
+    supported_kernels:
+      - fp64
+  - name: arithmetic.min
+    aggregate: true
+    supported_kernels:
+      - i16
+      - i32
+  - name: arithmetic.avg
+    aggregate: true
+    supported_kernels:
+      - fp64
+  - name: arithmetic_decimal.min
+    aggregate: true
+    supported_kernels:
+      - dec
+  - name: arithmetic_decimal.sum
+    aggregate: true
+    supported_kernels:
+      - dec
+`
+
+func TestGetLocalAggregateFunctionByInvocation(t *testing.T) {
+	header := makeAggregateTestHeader("v1.0", "/extensions/functions_arithmetic.yaml")
+	testsStr := `# basic
+avg((1,2,3)::fp32) = 2::fp64
+avg((1,2,3)::fp64) = 2::fp64
+min((20, -3, 1, -10, 0, 5)::i8) = -10::i8
+min((-32768, 32767, 20000, -30000)::i16) = -32768::i16
+min((-214748648, 214748647, 21470048, 4000000)::i32) = -214748648::i32
+sum((9223372036854775806, 1, 1, 1, 1, 10000000000)::i64) [overflow:ERROR] = <!ERROR>
+sum((2.5000007152557373046875, 7.0000007152557373046875, 0, 7.0000007152557373046875)::fp64) = 16.500002145767212::fp64
+`
+	decHeader := makeAggregateTestHeader("v1.0", "/extensions/functions_arithmetic_decimal.yaml")
+	decTestsStr := `# basic
+min((20, -3, 1, -10, 0, 5)::dec<2, 0>) = -10::dec<2, 0>
+min((-32768, 32767, 20000, -30000)::dec<5, 0>) = -32768::dec<5, 0>
+sum((2.5000007152557373046875, 7.0000007152557373046875, 0, 7.0000007152557373046875)::dec<23, 22>) = 16.5000021457672119140625::dec<38, 22>
+`
+
+	testResults := []struct {
+		name          string
+		expectedError string
+	}{
+		{"avg:fp32", "function variant not found"},
+		{"avg:fp64", ""},
+		{"min:i8", "function variant not found"},
+		{"min:i16", ""},
+		{"min:i32", ""},
+		{"sum:i64", "function variant not found"},
+		{"sum:fp64", ""},
+		{"min:dec", ""},
+		{"min:dec", ""},
```

Comment:
Could we add a test with an incompatible option (similar to the test done for scalar above)?